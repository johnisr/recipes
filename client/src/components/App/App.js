import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { fetchUser } from '../../actions/actions';

import Header from '../Header/Header';
import Landing from '../Landing/Landing';
import Dashboard from '../Dashboard/Dashboard';
import PrivateRoute from '../PrivateRoute/PrivateRoute';
import RecipeEdit from '../RecipeEdit/RecipeEdit';

const RecipeList = () => <div>Recipe List</div>;
const Recipe = () => <div>Specific Recipe</div>;

export class App extends Component {
  componentDidMount() {
    this.props.fetchUser();
  }
  render() {
    const { location } = this.props;
    return (
      <div className="container">
        <Header />
        <Route path="/" exact component={Landing} />
        <Route path="/recipes/:id" exact component={Recipe} />
        <Route path="/recipes" exact component={RecipeList} />
        <PrivateRoute
          location={location}
          path="/recipesNew"
          exact
          component={RecipeEdit}
        />
        <PrivateRoute
          location={location}
          path="/dashboard"
          exact
          component={Dashboard}
        />
        <PrivateRoute
          location={location}
          path="/recipes/:id/edit"
          exact
          component={RecipeEdit}
        />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  fetchUser: () => dispatch(fetchUser()),
});

export default withRouter(connect(null, mapDispatchToProps)(App));
