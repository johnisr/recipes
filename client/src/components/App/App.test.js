import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

test('should render correctly', () => {
  const wrapper = shallow(<App fetchUser={jest.fn()} getRecipes={jest.fn()} />);
  expect(wrapper).toMatchSnapshot();
});
