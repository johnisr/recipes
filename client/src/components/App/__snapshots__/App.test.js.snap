// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchUser={[Function]}
    getRecipes={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/recipes/:id"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/recipes"
/>,
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          location={undefined}
          path="/recipesNew"
/>,
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          location={undefined}
          path="/dashboard"
/>,
        <Connect(PrivateRoute)
          component={[Function]}
          exact={true}
          location={undefined}
          path="/recipes/:id/edit"
/>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/recipes/:id",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/recipes",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "location": undefined,
          "path": "/recipesNew",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "location": undefined,
          "path": "/dashboard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "location": undefined,
          "path": "/recipes/:id/edit",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/recipes/:id"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/recipes"
/>,
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            location={undefined}
            path="/recipesNew"
/>,
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            location={undefined}
            path="/dashboard"
/>,
          <Connect(PrivateRoute)
            component={[Function]}
            exact={true}
            location={undefined}
            path="/recipes/:id/edit"
/>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/recipes/:id",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/recipes",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "location": undefined,
            "path": "/recipesNew",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "location": undefined,
            "path": "/dashboard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "location": undefined,
            "path": "/recipes/:id/edit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
